llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -5025,10 +5026,41 @@ static Function *createOutlinedFunction(
       ParameterTypes.push_back(Arg->getType());
   }
 
+  auto BB = Builder.GetInsertBlock();
+  auto M = BB->getModule();
   auto FuncType = FunctionType::get(Builder.getVoidTy(), ParameterTypes,
                                     /*isVarArg*/ false);
-  auto Func = Function::Create(FuncType, GlobalValue::InternalLinkage, FuncName,
-                               Builder.GetInsertBlock()->getModule());
+  auto Func =
+      Function::Create(FuncType, GlobalValue::InternalLinkage, FuncName, M);
+
+  // If there's a DISubprogram associated with current function, then
+  // generate one for the outlined function.
+  if (Function *parentFunc = BB->getParent()) {
```

Comment:
Do we expect this to be an unconnected BB at any point?
Here, and below, please use LLVM style. This is not MLIR.

Reply:
I don't think BB will be unconnected. But even if that was to happen, the worst we will do is to drop debug bits as we are already doing. 

Thanks for pointing out the style bit. Corrected now.